xddq/ts2typebox
# WIP - add ability to parse indexed access types

- Noticed that indexed access types were not working and implemented a  first version of it supporting indexed access types with depth1 and depth2. Was to stupid to implement it with recursion yet, might be a task for another day :p. Main changes in this file [here](https://github.com/xddq/ts2typebox/pull/2/files#diff-2d6cde5bd1603d51af990557190f42976eba2d85f2a87556bc5535d0e18c3b2cR31)

- Added skeleton for jest tests

## How to test
- **test index access with "depth 2"**. Create types.ts with content

```
// Arbitrary example types

export type Address = {
  street: string;
  test: {
    a: string;
  };
};

export type Address2 = {
  street2: Address["test"]["a"];
};

```
- run `yarn build && yarn start`
- expected output

```
import { Type, Static } from '@sinclair/typebox'

export type Address = Static<typeof Address>
export const Address = Type.Object({
  street: Type.String(),
  test: Type.Object({
    a: Type.String()
  })
})

export type Address2 = Static<typeof Address2>
export const Address2 = Type.Object({
  street2: Type.String()
})

```

- **test index access with "depth 1"**
- create types.ts with content

```
// Arbitrary example types

export type Address = {
  street: string;
  test: {
    a: string;
  };
};

export type Address2 = {
  street2: Address["test"];
};

```

- run `yarn build && yarn start`
- expected output

```
import { Type, Static } from '@sinclair/typebox'

export type Address = Static<typeof Address>
export const Address = Type.Object({
  street: Type.String(),
  test: Type.Object({
    a: Type.String()
  })
})

export type Address2 = Static<typeof Address2>
export const Address2 = Type.Object({
  street2: Type.Object({
    a: Type.String()
  })
})

```

- dummy tests should pass, run `yarn build && yarn test`
